client/tailscale/authkey.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tailscale
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// Key represents a Tailscale auth key.
+type Key struct {
+	ID           string          `json:"id"`
+	Created      time.Time       `json:"created"`
+	Expires      time.Time       `json:"expires"`
+	Capabilities KeyCapabilities `json:"capabilities"`
+}
+
+// KeyCapabilities are the capabilities of an auth key.
+type KeyCapabilities struct {
+	Devices KeyDeviceCapabilities `json:"devices,omitempty"`
+}
+
+// KeyDeviceCapabilities are the device-related capabilities of an auth key.
+type KeyDeviceCapabilities struct {
+	Create KeyDeviceCreateCapabilities `json:"create"`
+}
+
+// KeyDeviceCreateCapabilities are the device creation capabilities of an auth key.
+type KeyDeviceCreateCapabilities struct {
+	Reusable      bool     `json:"reusable"`
+	Ephemeral     bool     `json:"ephemeral"`
+	Preauthorized bool     `json:"preauthorized"`
+	Tags          []string `json:"tags,omitempty"`
+}
+
+// Keys returns the list of auth keys for the current user.
```

Comment:
This endpoint will return both api keys and auth keys.  I think the only endpoint here that works exclusively with auth keys is key creation.  So we probably want to update documentation throughout (and maybe rename the file to keys.go or similar?)

Reply:
Hmm, from the docs and server-side code, it looked like everything only returned authkeys, but I now see that it also supports API keys, albeit not ideally right now (e.g. returns no metadata about them). I'll update the docs and code to be about all kinds of keys.

Reply:
Adjusted to reflect that the API can (mostly) handle all key types, PTAL.